zebrad/src/components/sync/downloads.rs

```diff
@@ -169,6 +182,16 @@ where
                     rsp = block_req => rsp?,
                 };
 
+                // Check if the block is already in the state.
+                // BUG: check if the hash is in any chain (#862).
+                // Depth only checks the main chain.
+                match state.oneshot(zs::Request::Depth(hash)).await {
+                    Ok(zs::Response::Depth(None)) => Ok(()),
+                    Ok(zs::Response::Depth(Some(_))) => Err("already present".into()),
```

Comment:
There's a nasty hack in the syncer code, where we do a string match on this error.

So we have to use a specific string here:
```suggestion
                    // TODO: use an error enum here, and update the sync AlreadyVerified check
                    Ok(zs::Response::Depth(Some(_))) => Err("AlreadyVerified { }".into()),
```

We'll clean up this code in #2339.

Reply:
Thanks, https://github.com/ZcashFoundation/zebra/pull/2327/commits/ec445d223912bd1006c98d2aa899fc73b1dc3931